### PR TITLE
LibWeb: Support displaying HTMLInputElement placeholder values

### PR DESCRIPTION
**Pull request by trflynn89**
[Screencast from 2022-11-30 22-06-58.webm](https://user-images.githubusercontent.com/5600524/205066234-4f0d2fb1-919c-4013-ab72-95089e1975b5.webm)

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp [hunk 1]
```diff
@@ -317,6 +317,47 @@ WebIDL::ExceptionOr<void> HTMLInputElement::set_value(String value)
...
+    auto placeholder = attribute(HTML::AttributeNames::placeholder);
+
+    if (placeholder.contains('\r') || placeholder.contains('\n')) {
+        StringBuilder builder;
+        for (auto ch : placeholder) {
+            if (ch != '\r' && ch != '\n')
+                builder.append(ch);
+        }
+        placeholder = builder.to_string();
+    }
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp:356]: "Strip these characters from the string" feels like something AK::String should be able to do itself, but maybe it's not that common. :thinking:

**Review by AtkinsSJ — APPROVED** [review]
Looks good to me. :^)

File: Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp:356]: Yeah, there's `replace`, but we have to do `str.replace(..).replace(..)...` when we want to replace multiple chars, creating a new string each time :(

This block was yoinked from `HTMLInputElement::value_sanitization_algorithm` so it's probably a nice abstraction to have sooner than later :yakstack++:

Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp:356]: Yup. Also `replace()` allocates a Vector to store all the positions, which is a bit :redagni:

**Comment by TobyAsE** [comment]
TIL ´::placeholder´ pseudo class exists

**Pull request merged**